### PR TITLE
New package: py-cdsapi

### PR DESCRIPTION
**Pull request by JSquar**
Package allows to access Climate Data Store for downloading climate and weather reanalysis data, e.g. ERA5 interim or ERA5-land

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-cdsapi/package.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyCdsapi(PythonPackage):
+    """The Climate Data Store Application Program Interface is a service providing programmatic access to CDS data. Get your UID and API key from the CDS portal at the address https://cds.climate.copernicus.eu/user and write it into the configuration file. Look at https://pypi.org/project/cdsapi/ for an example"""
+
+    homepage = "https://cds.climate.copernicus.eu"
+    url      = "https://files.pythonhosted.org/packages/source/c/cdsapi/cdsapi-0.2.3.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cdsapi/package.py]:
```suggestion
    url      = "https://pypi.io/packages/source/c/cdsapi/cdsapi-0.2.3.tar.gz"
```

File: var/spack/repos/builtin/packages/py-cdsapi/package.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+    version('0.2.3', '333b31ec263224399635db9b21a2e1a50cd73451f5179f8d967437e7c9161d9b')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cdsapi/package.py]:
```suggestion
    version('0.2.3', sha256='333b31ec263224399635db9b21a2e1a50cd73451f5179f8d967437e7c9161d9b')
```

File: var/spack/repos/builtin/packages/py-cdsapi/package.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+    depends_on('py-setuptools', type='build')
+    depends_on('py-requests', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cdsapi/package.py]:
```suggestion
    depends_on('py-requests@2.5.0:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-cdsapi/package.py (hunk 1 above)
Comment (JSquar) [reply var/spack/repos/builtin/packages/py-cdsapi/package.py]: Changed it.

File: var/spack/repos/builtin/packages/py-cdsapi/package.py (hunk 2 above)
Comment (JSquar) [reply var/spack/repos/builtin/packages/py-cdsapi/package.py]: Changed it.

File: var/spack/repos/builtin/packages/py-cdsapi/package.py (hunk 3 above)
Comment (JSquar) [reply var/spack/repos/builtin/packages/py-cdsapi/package.py]: Changed it.

**Pull request merged**